### PR TITLE
Refactoring decorator is_authorized

### PR DESCRIPTION
**Pull request by MaianaB**
<p><b>Feature/Bug description:</b> The method is_authorized verify if user be author and admin of institution.</p>

<p><b>Solution:</b> Added permissions and user method has_permission_entity to verify</p>

<p><b>TODO/FIXME:</b> n/a</p>

### CODE REVIEW

File: backend/admin.py [hunk 1]
```diff
@@ -484,6 +485,7 @@ def get(self):
         mayza_post_comIMG.put()
         add_comments_to_post(mayza, jorge, mayza_post_comIMG,
                              mayza.institutions[0], 1)
+        mayza.add_permissions(['edit_post', 'remove_post'], mayza_post_comIMG.key.urlsafe())
```
Comment (andreldsa) [inline backend/admin.py]: mayza_post_comIMG? tá estranho.

File: backend/handlers/post_handler.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 from utils import Utils
 from utils import login_required
-from utils import is_authorized
+from utils import has_permission_entity
```
Comment (andreldsa) [inline backend/handlers/post_handler.py]: ???

File: backend/handlers/post_handler.py [hunk 3]
```diff
@@ -56,19 +42,25 @@ def get(self, user, url_string):
...
-        """Set the informations about post."""
-        post.delete(user)
+        is_author = user.has_permission('remove_post', key)
+        is_admin = user.has_permission('remove_posts', post.institution.urlsafe())
+
+        if(is_author or is_admin):
+            """Set the informations about post."""
+            post.delete(user)
+        else:
+            Utils._assert(not(is_author or is_admin),
```
Comment (mayzabeel) [inline backend/handlers/post_handler.py]: Não ṕrecisa verificar novamente, se entrar no else já lança a exceção.

File: backend/handlers/post_handler.py [hunk 4]
```diff
@@ -56,19 +42,25 @@ def get(self, user, url_string):
...
+                          'User is not allowed to remove this post',
+                          NotAuthorizedException)
```
Comment (mayzabeel) [inline backend/handlers/post_handler.py]: essa aqui

File: backend/handlers/post_handler.py [hunk 5]
```diff
@@ -56,19 +42,25 @@ def get(self, user, url_string):
...
-    @is_authorized
     def delete(self, user, key):
         """Handle DELETE Requests."""
         """Get the post from the datastore."""
         obj_key = ndb.Key(urlsafe=key)
         post = obj_key.get()
 
-        """Set the informations about post."""
-        post.delete(user)
+        is_author = user.has_permission('remove_post', key)
```
Comment (mayzabeel) [inline backend/handlers/post_handler.py]: Fazer merge e verificar com new user.has_permission

**Comment by MaianaB** [comment]
Problems with merge master, new pull request
https://github.com/eciis/web/pull/586

**Pull request closed without merging**